native/src/routes/Pois.tsx

```diff
@@ -60,6 +76,23 @@ const Pois = ({
 }: PropsType): ReactElement => {
   const { t } = useTranslation('pois')
   const theme = useTheme()
+  const [selectedFeature, setSelectedFeature] = useState<Feature<Point> | null>(null)
+  const [userLocation, setUserLocation] = useState<number[] | undefined>(undefined)
+  const [featureLocations, setFeatureLocations] = useState<Feature<Point>[]>(prepareFeatureLocations(pois))
+
+  useEffect(() => {
+    if (!path) {
+      const featureLocations = prepareFeatureLocations(pois, userLocation)
+      const selectedPoiId = Number(route.params.selectedPoiId)
+      if (selectedPoiId) {
+        const currentFeature: Feature<Point> | undefined = featureLocations.find(
+          feature => feature.properties?.id === Number(route.params.selectedPoiId)
+        )
+        currentFeature && setSelectedFeature(currentFeature)
```

Comment:
I think this should always be called, even if no feature is selected/found since otherwise we may show an outdated selection
```suggestion
        setSelectedFeature(currentFeature)
```